Adding datatypes

JonasAls89 opened the pull request.

Commit:
adding pagination via connector_cli

Tested locally but I'm unsure whether the paging and/or system works as intended. When I change from and count in the system config i.e:

"list": {
      "id_expression": "{{ id }}",
      "method": "GET",
      "next_page_link": "$ENV(base_url)/v2/customer?from=3&count=2",
      "next_page_termination_strategy": ["empty-result", "same-next-page-link"],
      "payload_property": "values",
      "url": "{{ properties.datatype }}"
    },

I get that the pipe processes a total of 9 entities. When I go into the collect pipe and enable preview, I see that I get 3 duplicates now which is seen during the run. I compared two of them via: https://jsondiff.com/ . They were semantically identical.

My question is, why does this happen and it is intended? I can't gather why that would be the case.

Commit:
Update system.json

Commit:
Delete xxxxxx.conf.json

Commit:
Update system.json

Commit:
adding datatypes

Need to align with Tarjei on the model mapping and _id of each of the template pipes.

Commit:
Modifying datatypes

Commit:
removing unmapped datatypes

BaardBouvet commented on templates/customer.json:
```diff
@@ -1,135 +1,140 @@
 [
```
This is wrong, it should not contain this warning.

BaardBouvet commented on templates/customer.json:
```diff
@@ -1,135 +1,140 @@
+  },
+  {
+    "_id": "{{@ system @}}-{{@ datatype @}}-event-processed",
+    "comment": "WARNING! This pipe is generated from the template of the '{{@ datatype @}}' datatype and changes will be silently ignored during collapse. For more information see the connectorpy README.",
```
Same here

BaardBouvet commented on templates/customer.json:
```diff
@@ -1,135 +1,140 @@
+  },
+  {
+    "_id": "{{@ system @}}-{{@ datatype @}}-register-webhook",
+    "comment": "WARNING! This pipe is generated from the template of the '{{@ datatype @}}' datatype and changes will be silently ignored during collapse. For more information see the connectorpy README.",
```
And here

BaardBouvet commented on templates/customer.json:
```diff
@@ -1,135 +1,140 @@
+  },
+  {
+    "_id": "{{@ system @}}-{{@ datatype @}}-collect",
+    "comment": "WARNING! This pipe is generated from the template of the '{{@ datatype @}}' datatype and changes will be silently ignored during collapse. For more information see the connectorpy README.",
```
And here

BaardBouvet commented on templates/projectcategory.json:
```diff
@@ -0,0 +1,13 @@
+  "source": {
+    "operation": "list",
+    "properties": {
+      "datatype": "project/category"
```
I would call this property "url_part" or something to not confuse it with our datatype name which is the one in the pipe id. (But I'm not entirely sure, so we can discuss this one.)

JonasAls89 replied:
I'll let this stay as is for now then. But I get your point too.

BaardBouvet replied:
I agree with you. :)

Commit:
Update customer.json

hglad committed:
Use env var for service_url

axelborge committed:
Add customer category

axelborge committed:
Remove -

axelborge committed:
name fix

axelborge committed:
Add project

axelborge committed:
Rename

axelborge committed:
Update manifest.json

axelborge committed:
Add Vat

axelborge committed:
Support for orders

BaardBouvet commented on templates/order.json:
```diff
@@ -0,0 +1,185 @@
+    "source": {
+      "type": "embedded",
+      "entities": [{
+        "start": "~t2023-01-01T00:00:00Z"
```
Did you commit something wrong here? We can't hardcode a specific date + 2 months..

JonasAls89 commented on templates/order.json:
```diff
@@ -0,0 +1,185 @@
+              ]
+            ]
+          ],
+          ["add", "::operation", "get-raw"]
```
I get en Exection Error: "reason_why_stopped": "Entity or transform configuration specified an unknown operation 'get-raw'. Don't know what to do." Is get-raw supposed to be added in the system config as a route as well?

Commit:
changing order and invoice to work with required params and removing unsupported datatype vattype

Commit:
Adding additional datatypes after Docs update

All datatypes tested aside from orderline. The DTL works albeit it fails its run with:

"reason_why_stopped": "GET request failed for system 'xxxxxx' with url 'https://api.tripletex.io/v2/order/orderline/1720858'. The server returned an non-'OK' error code: 404\nThe response from the server was:\n\n{\n  \"status\" : 404,\n  \"code\" : 6000,\n  \"message\" : \"Object not found\",\n  \"link\" : \"https://www.tripletex.no/tripletex-api-2-0/\",\n  \"developerMessage\" : null,\n  \"validationMessages\" : null,\n  \"requestId\" : \"6617e312-5654-47b8-bb2f-0fb207241197\"\n}",

Commit:
Update customer.json

manually adding the customer config from main

Commit:
Adding orderlines logic and route for system to support parsing value payload_property

BaardBouvet committed:
Add collect of orderline

Commit:
changing postfix to be all and removing invoice plus warnings

Commit:
removing datatypes that did not work

BaardBouvet commented on templates/system.json:
```diff
@@ -4,31 +4,29 @@
     "delete": {
       "method": "DELETE",
-      "url": "{{ properties.datatype }}/{{ entity.id }}"
+      "url": "{{ properties.datatype }}/{{ properties.id }}"
```
We need to switch this to entity, that's what the share pipe expects.

JonasAls89 replied:
implemented.

BaardBouvet reviewed (changes requested):
We need to switch from properties to entity in the system, that's what the share pattern requires, so we need to use that in the lookup of orderlines as well.

Commit:
Changed system to use entity.id and orderline to propagate updates since webhooks does not seem to be working
